cmd/cluster/owner.go

```diff
@@ -0,0 +1,154 @@
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	ownerCmd.Flags().StringVarP(&ops.userName, "user-id", "u", ops.userName, "user to check the cluster owner on")
+
+	return ownerCmd
+}
+
+func newOwnerOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *ownerOptions {
+	return &ownerOptions{
+		IOStreams:     streams,
+		GlobalOptions: globalOpts,
+	}
+}
+
+func (o *ownerOptions) complete(cmd *cobra.Command, _ []string) error {
+
+	o.output = o.GlobalOptions.Output
+
+	return nil
+}
+
+func createConnection() (*sdk.Connection, error) {
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		// TODO: this should be tracked by bubbling an error up and, exposing so we can compare it here.
+		if strings.Contains(err.Error(), "Not logged in, run the") {
+			return nil, fmt.Errorf("Failed to create OCM connection: Authentication error, run the 'ocm login' command first.")
+		}
+		return nil, fmt.Errorf("Failed to create OCM connection: %v", err)
+	}
+	return connection, nil
+}
+
+func (o *ownerOptions) run() error {
+	connection, err := createConnection()
+	if err != nil {
+		return fmt.Errorf("could not createConnection: %w", err)
+	}
+
+	var (
+		accountName = o.userName
+		accountID   = ""
+	)
+
+	if accountName == "" {
+		fmt.Println("using the current user")
+		response, err := connection.AccountsMgmt().V1().CurrentAccount().Get().
+			Send()
+		if err != nil {
+			return fmt.Errorf("Can't send request: %v", err)
+		}
+		accountName = response.Body().Username()
+		accountID = response.Body().ID()
+	} else {
+		tmpl, err := template.New("test").Parse("id = '{{.}}' or username like '%{{.}}%' or email like '%{{.}}%'")
+		if err != nil {
+			return fmt.Errorf("could not parse template: %w", err)
+		}
+		var tpl bytes.Buffer
+
+		err = tmpl.Execute(&tpl, accountName)
+		if err != nil {
+			return fmt.Errorf("could not execute the template with the input %v: %w", accountName, err)
+		}
+		searchString := tpl.String()
+		fmt.Printf("the search query is '%s'\n", searchString)
+
+		response, err := connection.AccountsMgmt().V1().Accounts().List().Parameter("search", searchString).
+			Send()
+
+		if err != nil {
+			return fmt.Errorf("Can't send request: %v", err)
+		}
+
+		if response.Total() != 1 {
+			fmt.Println("the users that were found are:")
+			v1.MarshalAccountList(response.Items().Slice(), os.Stdout)
+			fmt.Println()
+			return fmt.Errorf("given username '%s' is not unique, found '%d' matches", accountName, response.Total())
+		}
+		accountID = response.Items().Get(0).ID()
+
+	}
+
+	if accountID == "null" || accountID == "" {
+		return fmt.Errorf("could not extract the accountID")
+	}
+
+	fmt.Printf("the user is '%s' and his ID is '%s'\n", accountName, accountID)
+
+	searchString := fmt.Sprintf("creator.id = '%s' and status != 'Deprovisioned' and status != 'Archived'", accountID)
```

Comment:
Should the static part of this be a string defined outside this function? or maybe a const?

Reply:
The code should still read well if the var name is descriptive.